tests/src/com.github.ben-manes.caffeine/caffeine/2.9.3/src/test/java/com_github_ben_manes_caffeine/caffeine/CaffeineTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com_github_ben_manes_caffeine.caffeine;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.CaffeineSpec;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.Scheduler;
+import com.google.common.testing.FakeTicker;
+import org.junit.jupiter.api.Test;
+
+import java.lang.ref.Cleaner;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CaffeineTest {
+    @Test
+    void testRefresh() {
+        LoadingCache<String, String> graphs = Caffeine.newBuilder().maximumSize(10_000).refreshAfterWrite(1, TimeUnit.MINUTES)
+                .build(key -> key.equals("Hello") ? "World" : "Universe");
+        assertThat(graphs.get("Hello")).isEqualTo("World");
+        assertThat(graphs.getAll(List.of("Hi", "Aloha"))).isEqualTo(Map.of("Hi", "Universe", "Aloha", "Universe"));
+    }
+
+    @Test
+    void testStatistics() {
+        Cache<String, String> graphs = Caffeine.newBuilder().maximumSize(10_000).recordStats().build();
+        graphs.put("Hello", "World");
+        assertThat(graphs.getIfPresent("Hello")).isEqualTo("World");
+    }
+
+    @Test
+    void testSpecification() {
+        CaffeineSpec spec = CaffeineSpec.parse("maximumWeight=1000, expireAfterWrite=10m, recordStats");
+        LoadingCache<String, String> graphs = Caffeine.from(spec).weigher((String key, String graph) -> graph.length())
+                .build(key -> key.equals("Hello") ? "World" : "Universe");
+        assertThat(graphs.get("Hello")).isEqualTo("World");
+    }
+
+    @Test
+    void testCleanup() {
+        LoadingCache<String, String> firstGraphs = Caffeine.newBuilder().scheduler(Scheduler.systemScheduler()).expireAfterWrite(10, TimeUnit.MINUTES)
+                .build(key -> key.equals("Hello") ? "World" : "Universe");
+        firstGraphs.put("Hello", "World");
+        assertThat(firstGraphs.getIfPresent("Hello")).isEqualTo("World");
```

Comment:
Is there a need for the `firstGraphs` here? It looks like `cleaner` only operates on `secondGraphs` instance ([here](https://github.com/oracle/graalvm-reachability-metadata/pull/388/files#diff-220cd86ae77195bc08dc7249346abd6ccbbbd4ac5f7b828f0bd266b128cc25baR56))

Reply:
- I looked through the relevant unit tests in my personal git design, and I realized that this involved a bug in the Tracing Agent of the old version of GraalVM CE For JDK11, which collected incorrect JSON entries and led to a refactoring. 
- I refactored `testCleanup()` on GraalVM CE For JDK17.